Fixed link to begin-with and end-results folders

oskardudycz opened the pull request:
As in the title: Fixed link to begin-with and end-results folders,

Fixed invalid `StorageTier` variable - it should be defined with the pascalCase, as:
- it's used with pascalCase later on,
- it's best practice to name variables with pascalCase
- other variables and parameters are named wih pascalCase

Applied also a bit of formatting.

oskardudycz committed:
Fixed link to end-results folder in ARM03 README

oskardudycz committed:
Fixed reference to end-with folder in ARM04

oskardudycz committed:
Fixed other links to exercises samples

ghost commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/signed)](https://cla.opensource.microsoft.com/microsoft/Learning-ARM?pullRequest=7) <br/>All CLA requirements met.

oskardudycz committed:
Fixed invalid `StorageTier` variable - it should be defined with the pascalCase, as:
- it's used with pascalCase later on,
- it's best practice to name variables with pascalCase
- other variables and parameters are named wih pascalCase

Applied also a bit of formatting.

oskardudycz committed:
Fixed typo `THe` into `The` in ARM07/README.md

fboucher commented:
Wonderful thank you for all those correction @oskardudycz
